src/adapters/scene_category.js

```diff
@@ -99,13 +126,16 @@ export default class SceneCategory {
   }
 
   highlightPoiMarker = (poi, highlight) => {
-    const marker = document.getElementById(getMarkerId(poi));
-    if (marker) {
-      if (highlight) {
-        marker.classList.add('active');
-      } else {
-        marker.classList.remove('active');
-      }
+    const pois = this.selectedPoi ? [this.selectedPoi] : [];
+    if (highlight && poi !== this.selectedPoi) {
+      pois.push(poi);
     }
+    this.map.getSource(ACTIVE_POIS_LAYER).setData(poisToGeoJSON(pois));
```

Comment:
```suggestion
  const pois = [
    ...this.selectedPoi ? [this.selectedPoi] : [],
    ...this.selectedPoi !== highlight && poi ? [poi] : [],
  ];
  this.map.getSource(ACTIVE_POIS_LAYER).setData(poisToGeoJSON(pois));
```

What do you think about using spreads operators ?
:dog2:  Such immutable. Much functional. :dog:

Reply:
If you really think that's more readable, why not.

Reply:
I used a similar form, but even simpler IMO.